specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/examples/BastionShareableLinkCreate.json

```diff
@@ -0,0 +1,47 @@
+{
+  "parameters": {
+    "api-version": "2019-08-01",
+    "subscriptionId": "subid",
+    "resourceGroupName": "rg1",
+    "bastionHostName": "bastionhosttenant",
+    "bslRequest": {
+      "vms": [
+        {
+          "vm": {
+            "id": "/subscriptions/subid/resourceGroups/rgx/providers/Microsoft.Compute/virtualMachines/vm1"
+          }
+        },
```

Comment:
can each entry be of VM schema  (array of VMs).  to skip one level of nesting .. looks simpler

vms": [
        {
            "id": "/subscriptions/subid/resourceGroups/rgx/providers/Microsoft.Compute/virtualMachines/vm1"
              ....other vm properties.......
          },
      {
            "id": "/subscriptions/subid/resourceGroups/rgx/providers/Microsoft.Compute/virtualMachines/vm2"
          }
 }

Reply:
current modelling forces them to reformat, rather than just doing toArray of list of VMs in hand.

---
In reply to: [367157794](https://github.com/Azure/azure-rest-api-specs/pull/7969#discussion_r367157794) [](ancestors = 367157794)

Reply:
I see the response has more properties - bsl and createdAt
it is up to you then.. 
Since they have to reformat, you may instead consider introducing custom VM schema with properties needed

vms": [
        {
            "id": "/subscriptions/subid/resourceGroups/rgx/providers/Microsoft.Compute/virtualMachines/vm1"
            "sharableLink" : "",
            "createdAt" : 
          },
      {
            "id": "/subscriptions/subid/resourceGroups/rgx/providers/Microsoft.Compute/virtualMachines/vm2"
            "sharableLink" : "",
            "createdAt" : 
          }
 }


---
In reply to: [367157964](https://github.com/Azure/azure-rest-api-specs/pull/7969#discussion_r367157964) [](ancestors = 367157964,367157794)

Reply:
I would prefer to keep it as is. This allows us to be able to add extra parameters associated with the link creation in future. For instance, we have plans to allow the user to specify a protocol or a port to connect to.